### PR TITLE
Add 'HideEmpy' section bool flag

### PR DESCRIPTION
**Pull request by katsadim**
### Description of your changes

`--Hide-empty` flag hides sections that have no content. 

For example:
```console
❯ ./terraform-docs md .
## Requirements

No requirements.
❯ ./terraform-docs md --hide-empty=true .

```

Only works for the following commands:
* md
* md document
* md table
* asciidoc
* asciidoc document
* asciidoc table

Fixes: https://github.com/terraform-docs/terraform-docs/issues/541

I have:

- [x] Read and followed terraform-docs' [contribution process].
- [x] All tests pass when I run `make test`.

### How has this code been tested

* Manually checked that the --help flag outputs info about `--hide-empty` only for the following cases:
  * md --help
  * md document --help
  * md table --help
  * asciidoc --help
  * asciidoc document --help
  * asciidoc table --help
* Added unit tests for asciidoc and markdown
* Manually checked that it runs

[contribution process]: https://git.io/JtEzg

### CODE REVIEW

**Comment by khos2ow** [comment]
Please also add `hide-empty: false` to [configuration's options](https://github.com/terraform-docs/terraform-docs/blob/master/docs/user-guide/configuration.md#options) as well.

File: internal/format/asciidoc_document_test.go [hunk 1]
```diff
@@ -119,6 +119,12 @@ func TestAsciidocDocument(t *testing.T) {
 				OutputValuesPath: "output_values.json",
 			},
 		},
+		"HideEmpty": {
```
[the remark this answers is not in the document]
Comment (katsadim) [reply internal/format/asciidoc_document_test.go]: addressed in 02eea3340066cd76a0cc7141525c45c90a4c3aba

File: internal/cli/config.go [hunk 2]
```diff
@@ -80,6 +83,7 @@ func DefaultConfig() *Config {
 		HeaderFrom:    "main.tf",
 		FooterFrom:    "",
 		Content:       "",
+		HideEmpty:     false,
```
[the remark this answers is not in the document]
Comment (katsadim) [reply internal/cli/config.go]: addressed in 02eea3340066cd76a0cc7141525c45c90a4c3aba

File: internal/print/settings.go [hunk 3]
```diff
@@ -23,6 +23,12 @@ type Settings struct {
 	// scope: Markdown
 	EscapeCharacters bool
 
+	// HideEmpty hide empty sections
+	//
+	// default: false
+	// scope: Global
```
[the remark this answers is not in the document]
Comment (katsadim) [reply internal/print/settings.go]: addressed in 02eea3340066cd76a0cc7141525c45c90a4c3aba

File: docs/user-guide/configuration.md [hunk 4]
```diff
@@ -99,6 +99,7 @@ settings:
   default: true
   description: false
   escape: true
+  hideempty: false
```
Comment (khos2ow) [inline docs/user-guide/configuration.md:102]: This should be `hide-empty`

Comment (katsadim) [reply docs/user-guide/configuration.md]: Oops apologies! You were clear in your previous comment https://github.com/terraform-docs/terraform-docs/pull/551#issuecomment-908654837. But for some reason  I saw `lockfile` down below and I figured that i had to remove the dash!

Just addressed in ffe84881adb239267543c852fa5e17ec14a3e279.

Comment (khos2ow) [reply docs/user-guide/configuration.md]: It's alright, no worries 🙂

**Review by khos2ow — APPROVED** [review]
Thanks @katsadim, lgtm!

**Pull request merged**